Apps/Admin/Server/Services/DelegationService.cs

```diff
@@ -0,0 +1,119 @@
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Admin.Server.Services
+{
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Threading.Tasks;
+    using AutoMapper;
+    using FluentValidation.Results;
+    using HealthGateway.Admin.Common.Constants;
+    using HealthGateway.Admin.Common.Models;
+    using HealthGateway.Admin.Server.Validations;
+    using HealthGateway.Common.Constants;
+    using HealthGateway.Common.Data.ErrorHandling;
+    using HealthGateway.Common.Data.ViewModels;
+    using HealthGateway.Common.Factories;
+    using HealthGateway.Common.Models;
+    using HealthGateway.Common.Services;
+    using HealthGateway.Database.Delegates;
+    using HealthGateway.Database.Models;
+    using Microsoft.Extensions.Configuration;
+
+    /// <inheritdoc/>
+    public class DelegationService : IDelegationService
+    {
+        private const string DelegationConfigSection = "Delegation";
+        private const string MaxDependentAgeKey = "MaxDependentAge";
+        private readonly IPatientService patientService;
+        private readonly IResourceDelegateDelegate resourceDelegateDelegate;
+        private readonly IMapper autoMapper;
+        private readonly int maxDependentAge;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DelegationService"/> class.
+        /// </summary>
+        /// <param name="configuration">The injected configuration provider.</param>
+        /// <param name="patientService">The injected patient service.</param>
+        /// <param name="resourceDelegateDelegate">The injected resource delegate delegate.</param>
+        /// <param name="autoMapper">The injected automapper provider.</param>
+        public DelegationService(
+            IConfiguration configuration,
+            IPatientService patientService,
+            IResourceDelegateDelegate resourceDelegateDelegate,
+            IMapper autoMapper)
+        {
+            this.patientService = patientService;
+            this.resourceDelegateDelegate = resourceDelegateDelegate;
+            this.autoMapper = autoMapper;
+            this.maxDependentAge = configuration.GetSection(DelegationConfigSection).GetValue(MaxDependentAgeKey, 12);
+        }
+
+        /// <inheritdoc/>
+        public async Task<DelegationResponse> GetDelegationInformationAsync(string phn)
+        {
+            // Get dependent patient information
+            RequestResult<PatientModel> dependentPatientResult = await this.patientService.GetPatient(phn, PatientIdentifierType.Phn).ConfigureAwait(true);
```

Comment:
This can return several possible unsuccessful statuses that we should map to appropriate responses.

Reply:
Will be handled in AB#15139.